tests/foreman/api/test_syncplan.py

```diff
@@ -493,43 +494,102 @@ def test_positive_repeatedly_add_remove(self):
+            should be validated (e.g. package, erratum, puppet_module)
+        :param bool after_sync: Specify whether you perform validation before
+            synchronization procedure is happened or after
+        :param int max_attempts: That value is basically introduced for slow
+            systems when user sure that sync procedure can take more than 300
+            seconds (30*10 where 10 is a default value for max_attempts
+            variable)
+
+        """
+        for i in range(max_attempts):
+            try:
+                for content in content_types:
+                    if after_sync:
+                        self.assertGreater(repo.content_counts[content], 0)
+                    else:
+                        self.assertFalse(repo.content_counts[content])
+                break
+            except AssertionError:
+                sleep(30)
+                repo = repo.read()
+        else:
+                raise AssertionError(
+                    'Repository contains invalid number of content entities')
+
     @run_only_on('sat')
     @tier2
     def test_positive_synchronize_product(self):
-        """Create a sync plan with one product and sync it.
+        """Create a sync plan with one product and sync it automatically.
 
-        @Assert: A sync plan is created with one product and product can be
-        synchronized.
+        @Assert: Product is synchronized successfully.
 
         @Feature: SyncPlan
         """
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow(),
+        ).create()
+        product = entities.Product(organization=self.org).create()
+        repo = entities.Repository(product=product).create()
+        self.validate_repo_content(
+            repo, ['erratum', 'package', 'package_group'], after_sync=False)
+        sync_plan.add_products(data={'product_ids': [product.id]})
+        self.validate_repo_content(
+            repo, ['erratum', 'package', 'package_group'])
 
-    @stubbed()
     @run_only_on('sat')
     @tier2
     def test_positive_synchronize_products(self):
-        """Create a sync plan with two products and sync them.
+        """Create a sync plan with multiple products and sync them afterwards
+        automatically.
 
-        @Assert: A sync plan is created with one product and products can be
-        synchronized.
+        @Assert: Multiple Products are synchronized successfully.
 
         @Feature: SyncPlan
         """
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow(),
+        ).create()
+        products = [
+            entities.Product(organization=self.org).create()
+            for _ in range(random.randint(3, 5))
+        ]
+        repos = [
+            entities.Repository(product=product).create()
+            for product in products
```

Comment:
how about
```python
        repos = [
            entities.Repository(product=product).create()
            for product in products
            for _ in range(randint(2, 3))
        ]
```
It will add some more coverage - you'll have not just multiple products, but multiple repos per each product too. So the test will also verify that all the repos from each product recieved the content and no more changed in test are needed :).